src/cargo/core/resolver/mod.rs

```diff
@@ -1287,15 +1287,17 @@ impl Context {
                        .entry(id.name().to_string())
                        .or_insert_with(HashMap::new)
                        .entry(id.source_id().clone())
-                       .or_insert_with(Vec::new);
+                       .or_insert_with(||Rc::new(Vec::new()));
         if !prev.iter().any(|c| c == summary) {
             self.resolve_graph.push(GraphNode::Add(id.clone()));
             if let Some(link) = summary.links() {
                 ensure!(self.links.insert(link.to_owned(), id.clone()).is_none(),
                 "Attempting to resolve a with more then one crate with the links={}. \n\
                  This will not build as is. Consider rebuilding the .lock file.", link);
             }
-            prev.push(summary.clone());
+            let mut inner: Vec<_> = (**prev).clone();
+            inner.push(summary.clone());
+            *prev = Rc::new(inner);
```

Comment:
If we're optimizing for clones here, I wonder if this'd be better managed through an `RcList` perhaps?

Reply:
I was thinking of that, but we very often iter over the activated items. Specifically `is_active` and `prev_active`. So it would be a pretty invasive change. I can give it a try if you want.

Also before we cloned every `vec` for every clone of Context for `# clone vecs = O(# activated * ticks)`. Now we are cloning a `vec` each time we activate for `# clone vecs = O(# activated)`. If I am miss reading it please correct me.

Reply:
Ah an excellent point!